src/platform/user/authentication/components/CreateAccountLink.jsx

```diff
@@ -1,19 +1,40 @@
 import React from 'react';
-import { CSP_CONTENT } from '../constants';
+import PropTypes from 'prop-types';
 import recordEvent from 'platform/monitoring/record-event';
 import * as authUtilities from 'platform/user/authentication/utilities';
+import { CSP_CONTENT, AUTH_EVENTS } from '../constants';
 
-function signupHandler(loginType) {
-  recordEvent({ event: `register-link-clicked-${loginType}` });
+function signupHandler(loginType, eventBase) {
+  recordEvent({ event: `${eventBase}-${loginType}` });
 }
 
-export default ({ csp, isDisabled }) => (
-  <a
-    href={authUtilities.signupUrl(csp)}
-    className={`vads-c-action-link--blue vads-u-padding-y--2p5 vads-u-width--full ${csp}`}
-    disabled={isDisabled}
-    onClick={() => signupHandler(csp)}
-  >
-    Create an account with {CSP_CONTENT[csp].COPY}
-  </a>
-);
+export default function Link({ csp, isDisabled, type = 'create' }) {
+  const { href, children, eventBase } = {
+    href:
+      type !== 'create'
+        ? authUtilities.sessionTypeUrl({ type: csp })
+        : authUtilities.signupUrl(csp),
+    children:
+      type !== 'create'
+        ? `Sign in with ${CSP_CONTENT[csp].COPY} account`
+        : `Create an account with ${CSP_CONTENT[csp].COPY}`,
+    eventBase: type !== 'create' ? AUTH_EVENTS.LOGIN : AUTH_EVENTS.REGISTER,
```

Comment:
If this component can now be used for login and signup we should change the name of the component to be more generic

Reply:
Resolved.  Created a `TYPE_LINKS` constant